File: src/main/java/net/frozenorb/potpvp/util/menu/buttons/BackButton.java
package net.frozenorb.potpvp.util.menu.buttons;

import net.frozenorb.potpvp.util.menu.Button;
import net.frozenorb.potpvp.util.menu.Menu;
import lombok.AllArgsConstructor;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.event.inventory.ClickType;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
public class BackButton extends Button {

    private final Menu back;

    @Override
    public Material getMaterial(Player player) {
        return Material.REDSTONE;
    }

    @Override
    public byte getDamageValue(Player player) {
        return 0;
    }

    @Override
    public String getName(Player player) {
        return "§cGo back";
    }

    @Override
    public List<String> getDescription(Player player) {
        return new ArrayList<>();
    }

    @Override
    public void clicked(Player player, int i, ClickType clickType) {
        Button.playNeutral(player);
        Menu.getCurrentlyOpenedMenus().get(player.getUniqueId()).onClose(player);
        this.back.openMenu(player);
    }

}

File: src/main/kotlin/net/frozenorb/potpvp/kt/menu/buttons/BackButton.kt
package net.frozenorb.potpvp.kt.menu.buttons

import net.frozenorb.potpvp.kt.menu.Button
import org.bukkit.Material
import org.bukkit.event.inventory.ClickType
import org.bukkit.entity.Player
import org.bukkit.inventory.InventoryView
import java.util.ArrayList

class BackButton(private val callback: (Player) -> Unit) : Button() {

    override fun getMaterial(player: Player): Material {
        return Material.BED
    }

    override fun getName(player: Player): String {
        return "§cGo back"
    }

    override fun getDescription(player: Player): List<String> {
        return ArrayList()
    }

    override fun clicked(player: Player, i: Int, clickType: ClickType, view: InventoryView) {
        playNeutral(player)
        player.closeInventory()

        callback.invoke(player)
    }

}